### ISSUE TITLE
2D Arrow/Fireball does not travel

### ISSUE THREAD

**Issue by wtfcolt** [suriyun-production/mmorpg-kit-docs#163]
in 1.51, 2d arrow/fireball does not have range. player will walk to melee range to cast. Arrows do not travel. In picture, archers shoot but arrows do not travel to player. They appear on the archer for a second then vanish.

![image](https://user-images.githubusercontent.com/1743424/71302423-c700fd80-2378-11ea-873c-446710782481.png)

**Comment by insthync** [comment]
[MMORPGKIT-FixingMissileHitNonTarget.zip](https://github.com/suriyun-production/mmorpg-kit-docs/files/3991102/MMORPGKIT-FixingMissileHitNonTarget.zip)
Fix by this package

**Comment by wtfcolt** [comment]
Bug still happening. Default 1.51 + fix posted above

https://youtu.be/j6ZQfgucBCk

**Comment by insthync** [comment]
Try changes codes in `MissileDamageEntity.cs` from line 167 to
```
            if (other.layer != 1 &&
                other.layer != 2 &&
                other.layer != gameInstance.characterLayer &&
                other.layer != gameInstance.itemDropLayer &&
                !gameInstance.NonTargetLayersValues.Contains(other.layer))
            {
                if (explodeDistance > 0f)
                {
                    // Explode immediately when hit something
                    Explode();
                }
                Destroy(gameObject, destroyDelay);
                destroying = true;
                return;
            }
```

**Comment by wtfcolt** [comment]
Working for arrows, thanks! 

Firebolt still requires melee range however.

**Comment by insthync** [comment]
Fixed in 1.52
